### PR TITLE
Fix flakey month testing

### PR DESCRIPTION
**Pull request by TurnerRiley**
A few tests introduced by [this PR](https://github.com/code-dot-org/code-dot-org/pull/51309) were showing flakey behavior due to time edge cases (as discussed in [this slack thread](https://codedotorg.slack.com/archives/C03CM903Y/p1685450383887159)). A quick fix-forward was made [here](https://github.com/code-dot-org/code-dot-org/pull/52139) but the problem seemed to persist.

This PR simplifies all logic using `month` related to the original PR and instead uses 30 days as a replacement to provide consistency. Given that the purpose of the [original PR](https://github.com/code-dot-org/code-dot-org/pull/51309) was to just prevent users from updating certain details too close to the start of a workshop starting, it seems safe to just average that time out to just 30 days. With these changes, the flakey tests seem to be consistently passing!

## Links
Original PR: [here](https://github.com/code-dot-org/code-dot-org/pull/51309)
Slack thread: [here](https://codedotorg.slack.com/archives/C03CM903Y/p1685450383887159)
First quick fix: [here](https://github.com/code-dot-org/code-dot-org/pull/52139)

## Testing story
Local testing and upating unit tests.

## Follow-up work
Monitor failing tests to see if this one pops up again.

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by mgc1194 — APPROVED** [review]

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx [hunk 1]
```diff
@@ -380,14 +380,16 @@ export class WorkshopForm extends React.Component {
...
+  // in workshop_controller.rb).
   sessionStartsWithinMonth = session => {
     const today = this.props.today;
     const workshopDate = new Date(session.date);
     const monthBeforeWorkshopDate = new Date(
       workshopDate.getFullYear(),
-      workshopDate.getMonth() - 1,
-      workshopDate.getDate()
+      workshopDate.getMonth(),
+      workshopDate.getDate() - 30
```
Comment (davidsbailey) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:392]: hmm I am not sure this does what we want:
Screenshot 2023-05-31 at 10 44 23 AM

Comment (davidsbailey) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:392]: apologies, this does look like it is working, I just didn't realize the months are 0-indexed (while the days are 1-indexed). this makes it look like it is working properly to me: 
Screenshot 2023-05-31 at 10 48 43 AM

**Review by davidsbailey — APPROVED** [review]
looks good, thanks for the quick fix Turner!

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:392]: I think it is right? The month `05` maps to June (0-indexed), and May has 31 days so June 15 - 30 days is May 16?

Comment (davidsbailey) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:392]: makes sense, thanks for thinking it through!

**Pull request merged**